### ISSUE TITLE
when will pre_eval support samples_per_gpu > 1 ?

### ISSUE THREAD

**Issue by fangwudi** [open-mmlab/mmsegmentation#1089]
I noticed that in api/test.py it says pre_eval only samples_per_gpu=1 valid now. when wiill pre_eval support samples_per_gpu > 1 ?

**Comment by MengzhangLI** [comment]
Hi, we plan to do that but no clear time schedule. Because batch size > 1 means when images sizes are different, cropping and padding is inevitable, which also makes metric calculation more complicated. 

If you have some suggestions, please comment here, looking forward to your reply.

Best,

**Comment by fangwudi** [comment]
> inevitable

Maybe you can complete 'batchsize > 1 and images sizes are resized as the same' first?

**Comment by MengzhangLI** [comment]
> 

Yes, but it  would affect metric calculation because input images are not totally the same.

**Comment by Bonnie-gift** [comment]
Hi, I see you fork the project "Aerial template matching". Have you done the experiment? I have done the experiment but there are some questions wanting to communicate with you. If you have time, please contact me! My e-mail address [zhangjy991108@163.com](mailto:zhangjy991108@163.com)

**Comment by fangwudi** [comment]
> 

just forked. not tried yet.
